app/src/main/java/org/simple/clinic/home/HomeScreenController.kt

```diff
@@ -26,11 +32,49 @@ class HomeScreenController @Inject constructor(
         .replay()
 
     return Observable.merge(
-        currentFacility(replayedEvents),
+        screenSetup(replayedEvents),
         changeFacility(replayedEvents))
   }
 
-  private fun currentFacility(events: Observable<UiEvent>): Observable<UiChange> {
+  private fun screenSetup(events: Observable<UiEvent>): Observable<UiChange> {
+    val facilityStream = facilityStream(events)
+    val overdueAppointmentCountStream = overdueAppointmentCountStream(facilityStream)
+
+    val homeFacilityNameStream = facilityStream
+        .map { facility -> { ui: Ui -> ui.setFacility(facility.name) } }
+
+    val homeOverdueAppointmentCountStream = homeOverdueAppointmentCountStream(overdueAppointmentCountStream)
+
+    return Observable.merge(homeFacilityNameStream, homeOverdueAppointmentCountStream)
+  }
+
+  private fun homeOverdueAppointmentCountStream(overdueAppointmentCountStream: Observable<List<Int>>): Observable<(Ui) -> Unit>? {
+    return overdueAppointmentCountStream
+        .map { overdueCountList ->
+          { ui: Ui ->
+            val count = overdueCountList.size
+            if (count > 0) {
+              ui.showOverdueAppointmentCount(overdueCountList.size)
+            } else {
+              ui.removeOverdueAppointmentCount()
+            }
+          }
+        }
+  }
+
+  private fun overdueAppointmentCountStream(facilityStream: Observable<Facility>): Observable<List<Int>> {
+    return facilityStream
+        .flatMap { currentFacility ->
+          appointmentRepository.overdueAppointmentsCount(
+              since = LocalDate.now(userClock),
+              facility = currentFacility
+          )
+        }
+        .replay()
+        .refCount()
```

Comment:
This stream does not need to be shared since it has only one consumer.

Reply:
done